examples/demo-app/src/cloud-providers/fsq-studio/fsq-studio-provider.js

```diff
@@ -0,0 +1,177 @@
+  return data;
+}
+
+function convertMapToFSQStudio(map) {
+  return {
+    description: map.info.description,
+    name: map.info.title,
+    importSource: STUDIO_KEPLER_GL_IMPORT_SOURCE,
+    latestState: {
+      data: map
+    }
+  };
+}
+
+export default class FsqStudioProvider extends Provider {
+  constructor({clientId, authDomain, apiURL, userMapsURL}) {
+    super({name: NAME, displayName: DISPLAY_NAME, icon: FSQIcon});
+    this.icon = FSQIcon;
+    this.appName = APP_NAME;
+    this.apiURL = apiURL;
+
+    const redirect_uri = window.location.origin + window.location.pathname;
+
+    this._auth0 = new Auth0Client({
+      domain: authDomain,
+      clientId: clientId,
+      scope: STUDIO_AUTH_SCOPE,
+      authorizationParams: {
+        redirect_uri,
+        audience: STUDIO_AUTH_AUDIENCE
+      },
+      cacheLocation: 'localstorage'
+    });
+
+    // the domain needs to be passed as input param
+    this._folderLink = userMapsURL;
+  }
+
+  hasPrivateStorage() {
+    return STUDIO_PRIVATE_STORAGE_ENABLED;
+  }
+
+  async getUser() {
+    return this._auth0.getUser();
+  }
+
+  async login() {
+    return this._auth0.loginWithPopup();
+  }
+
+  async logout() {
+    return this._auth0.logout();
+  }
+
+  async uploadMap({mapData, options = {}}) {
+    // TODO: handle replace
+    // const mode = options.overwrite ? 'overwrite' : 'add';
+    const {map, thumbnail} = mapData;
+
+    const {title = '', description = ''} = map.info;
+    const headers = await this.getHeaders();
+    const payload = {
+      name: title,
+      description,
+      importSource: STUDIO_KEPLER_GL_IMPORT_SOURCE,
+      latestState: {
+        data: map
+      }
+    };
+
+    const mapResponse = await fetch(`${this.apiURL}/v1/maps`, {
+      method: 'POST',
+      headers,
+      body: JSON.stringify(payload)
+    });
+
+    const createMap = await mapResponse.json();
+
+    await fetch(`${this.apiURL}/v1/maps/${createMap.id}/thumbnail`, {
+      method: 'PUT',
+      headers: {
+        ...headers,
+        'Content-Type': 'image/png'
+      },
+      body: thumbnail
+    });
+
+    return map;
+  }
+
+  async listMaps() {
+    const headers = await this.getHeaders();
+    const response = await fetch(
+      `${this.apiURL}/v1/maps?importSource=${STUDIO_KEPLER_GL_IMPORT_SOURCE}`,
+      {
+        method: 'GET',
+        mode: 'cors',
+        headers
+      }
+    );
+    const data = await response.json();
+    return data.items.map(convertFSQModelToMapItem);
+  }
+
+  async downloadMap(loadParams) {
+    const {mapId} = loadParams;
+    if (!mapId) {
+      return Promise.reject('No Map is was provider as part of loadParams');
+    }
+    const headers = await this.getHeaders();
+
+    const response = await fetch(`${this.apiURL}/v1/maps/${mapId}`, {
+      method: 'GET',
+      headers
+    });
+
+    const map = await response.json();
+
+    return Promise.resolve({
+      map: extractMapFromStudioResponse(map),
+      format: KEPLER_FORMAT
```

Comment:
`KEPLER_FORMAT` string could probably be made a little longer / more self descriptive. There are a lot of different types of "formats" being managed by kepler.gl